File: outlines/08-functions-in-js/workshop/calculator.js
// calculator function
function calculator(operation, argument1, argument2) {
  if(operation === "add") {
    return argument1 + argument2;
  } else if(operation === "subtract") {
    return argument1 - argument2;
  } else if(operation === "multiply" ) {
    return argument1 * argument2;
  } else if(operation === "divide") {
    // division by zero
    if(argument2 === 0) {
      return -1;
    }

    return argument1 / argument2;
  }
}

// invoke function
console.log(calculator("multiply", 4, 2))
console.log(calculator("add", 4, 2))
console.log(calculator("divide", 4, 2))
console.log(calculator("subtract", 4, 2))

File: outlines/27-object-oriented-programming-using-ruby/exercises/calculator.rb
# defines a new calculator class
class Calculator
  # sets the class method sum
  def self.add(arg1, arg2)
    arg1 + arg2
  end

  # sets the class method difference
  def self.subtract(arg1, arg2)
    arg1 - arg2
  end

  # sets the class method product
  def self.multiply(arg1, arg2)
    arg1 * arg2
  end

  # sets the class method quotient
  def self.divide(arg1, arg2)
    arg1 / arg2
  end
end

# test our Calculator class methods
puts Calculator.add(1, 2) # 3
puts Calculator.subtract(5, 4) # 1
puts Calculator.multiply(3, 2) # 6
puts Calculator.divide(10, 2) # 5